### PR TITLE
Release notes for simulation as part of the Documentation.

### PR DESCRIPTION
**Pull request by jcapriot**
This PR adds release notes for the Simulation branch to the SimPEG documentation itself.

I would like to get other release notes in there as well from past versions, but that's a much lower priority. Moving forward I would like release notes like this.

Also, there's a few bug fixes/deprecation updates in here.

### CODE REVIEW

**Comment by rowanc1** [comment]
The release notes look amazing. Lots of good work and thought going into this. :)
Nice work all!!

**Comment by jcapriot** [comment]
it might be easier for some of you to read it through this link, https://github.com/simpeg/simpeg/blob/sim-release-notes/docs/content/release/0.14.0-notes.rst

File: AUTHORS.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@
 - Brendan Smithyman, (`@bsmithyman <https://github.com/bsmithyman/>`_)
 - Gudni Rosenkjaer, (`@grosenkj <https://github.com/grosenkj/>`_)
 - Dom Fournier, (`@fourndo <https://github.com/fourndo/>`_)
+- Joseph Capriotti, (`@jcapriot <https://github.com/jcapriot`)
```
Comment (lheagy) [inline AUTHORS.rst:7]: 🎉

File: docs/content/release/0.14.0-notes.rst [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+
+This ``Data`` class is now also the object that is returned from:
+
+.. code-block:: python
+
+    data = sim.make_synthetic_data(
+        m, relative_error=0.05, noise_floor=0.0, f=None, add_noise=True
+    )
+
+The ``L2DataMisfit`` class
```
Comment (lheagy) [inline docs/content/release/0.14.0-notes.rst:158]: Do we want this to be just `DataMisfit` ? (I know all that we have at the moment is the L2DataMisfit, but the concept would be the same for any generic data misfit)

File: docs/content/release/0.14.0-notes.rst [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+Combining to produce over 500 commits (listed in no particular order):
+
+* @lheagy
+* @jcapriot
+* @fourndo/@domfournier
+* @sgkang
+* @dccowan
+* @JKutt
+* @prisae
+* @craigmillernz
```
Comment (lheagy) [inline docs/content/release/0.14.0-notes.rst:406]:
```suggestion
* `@lheagy <https://github.com/lheagy>`__
* `@jcapriot <https://github.com/jcapriot>`__
* `@fourndo <https://github.com/fourndo>`__  / `@domfournier <https://github.com/domfournier>`__
* `@sgkang <https://github.com/sgkang>`__
* `@dccowan <https://github.com/dccowan>`__
* `@JKutt <https://github.com/JKutt>`__
* `@prisae <https://github.com/prisae>`__
* `@craigmillernz <https://github.com/craigmillernz>`__
```

File: docs/content/release/0.14.0-notes.rst (hunk 2 above)
Comment (jcapriot) [reply docs/content/release/0.14.0-notes.rst:158]: Are you referring to just the title of this section? If so I'd be happy with that.

File: docs/content/release/0.14.0-notes.rst (hunk 3 above)
Comment (jcapriot) [reply docs/content/release/0.14.0-notes.rst:406]: All the links!

**Comment by jcapriot** [comment]
> Closed



> This looks really solid @jcapriot!! I just left a couple minor comments. One other thought - do we want to have an "Issues Closed" section similar to the "PRs" to keep track of when things were addressed?

@lheagy I was certainly thinking of adding that in, but didn't because I just was not sure of all of the issues that it closes :).

**Comment by lheagy** [comment]
Thats fair, out issues are a bit all-over the place. We could hold off on adding for now and do a pass on the issues after we merge it in.

File: docs/content/release/0.14.0-notes.rst [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+
+    uncertainty = survey.std * np.abs(survey.dobs) + survey.eps
+
+We now have updated the names to be clearer and more in line with what we would
+naturally expect, which is accessed from `data.standard_deviation`. The value
+that is returned from this is now defined as:
+
+.. code-block:: python
+
+    data.standard_deviation = (
```
Comment (sgkang) [inline docs/content/release/0.14.0-notes.rst:145]: may be worthwhile to mention that user can set `data.standard_deviation` as well?
One may have a fancier way to define uncertainty.

**Comment by sgkang** [comment]
@jcapriot this is amazing work leading us to a right direction! 
Now, it seems a relevant time to update some of my old codes still uses old framework (`Problem`?), and I am motivated! Thanks again for this amazing work!

File: AUTHORS.rst (hunk 1 above)
Comment (prisae) [reply AUTHORS.rst:7]: I think the final `` ` `` should be a ``>`_`` in your name and John's name

Comment (jcapriot) [reply AUTHORS.rst:7]: yup...

File: docs/content/release/0.14.0-notes.rst [hunk 5]
```diff
@@ -0,0 +1,406 @@
+.. _0.14.0_notes:
+
+===========================
+SimPEG 0.14.0 Release Notes
```
Comment (prisae) [inline docs/content/release/0.14.0-notes.rst:4]: Looks awesome @jcapriot !

File: docs/content/release/0.14.0-notes.rst (hunk 2 above)
Comment (lheagy) [reply docs/content/release/0.14.0-notes.rst:158]: Yep, just the title 
```suggestion
The ``DataMisfit`` class
```

File: docs/content/release/0.14.0-notes.rst (hunk 4 above)
Comment (jcapriot) [reply docs/content/release/0.14.0-notes.rst:145]: definitely valid, I'll add in something.

**Comment by JKutt** [comment]
Awesome stuff here @jcapriot ! Reads well and looks good to me.

**Pull request merged**